packages/amplify_datastore/example/integration_test/utils/wait_for_expected_event_from_hub.dart

```diff
@@ -0,0 +1,55 @@
+import 'dart:async';
+
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+import 'package:amplify_flutter/amplify_flutter.dart';
+
+class WaitForExpectedEventFromHub<T extends HubEventPayload> {
+  final Completer<T> _completer = Completer();
+  late StreamSubscription hubSubscription;
+  final Function eventMatcher;
+  final String eventName;
+  Duration timeout;
+
+  WaitForExpectedEventFromHub({
+    required this.eventMatcher,
+    required this.eventName,
+    Duration timeout = const Duration(seconds: 20),
+  }) : this.timeout = timeout;
+
+  Future<T> start() {
+    hubSubscription = Amplify.Hub.listen([HubChannel.DataStore], (event) {
+      if (event.eventName == this.eventName) {
+        if (this.eventMatcher(event.payload)) {
+          hubSubscription.cancel();
+          _completer.complete(event.payload as T);
+        }
+      }
+    });
+    startTimeout();
```

Comment:
nit: Same comment as in setup_utils.dart, but can we get rid of this utility and just call `.timeout` on the future in line 29?